src/test/java/io/debezium/connector/yugabytedb/YugabyteDBSnapshotTest.java

```diff
@@ -1018,6 +1019,93 @@ public void snapshotShouldBeCompletedOnParentIfSplitHappenedAfterStreamCreation(
+        final int STREAM_RECORDS = 10;
+
+        createTablesInColocatedDB(true);
+
+        // Insert records with empty name - these will be filtered out
+        String formatInsertString = "INSERT INTO public.test_1(id, name) VALUES (%d, '');";
+        TestHelper.executeBulk(formatInsertString, SNAPSHOT_RECORDS, DEFAULT_COLOCATED_DB_NAME);
+
+        // Insert 10 records with non-empty name - these should pass the filter in snapshot
+        for (int i = SNAPSHOT_RECORDS; i < SNAPSHOT_RECORDS + STREAM_RECORDS; i++) {
+            TestHelper.executeInDatabase(
+                "INSERT INTO public.test_1(id, name) VALUES (" + i + ", 'Shishir');",
+                DEFAULT_COLOCATED_DB_NAME
+            );
+        }
+
+        // Create CDC stream and configure connector with filter
+        String dbStreamId = TestHelper.getNewDbStreamId(
+            DEFAULT_COLOCATED_DB_NAME, "test_1", true, true
+        );
+        Configuration.Builder configBuilder = TestHelper.getConfigBuilder(
+            DEFAULT_COLOCATED_DB_NAME, "public.test_1", dbStreamId
+        );
+        configBuilder.with(YugabyteDBConnectorConfig.SNAPSHOT_MODE, snapshotMode.getValue());
+        
+        // Configure Debezium Filter SMT to filter records with non-empty name
+        configBuilder.with("transforms", "filter");
+        configBuilder.with("transforms.filter.type", "io.debezium.transforms.Filter");
+        configBuilder.with("transforms.filter.language", "jsr223.groovy");
+        configBuilder.with("transforms.filter.condition",
+            "(value.op == 'r' && value.after.name?.value?.trim() != null && value.after.name?.value?.trim() != '') || " +
+            "(value.op == 'c' && value.after.name?.value?.trim() != null && value.after.name?.value?.trim() != '')"
+        );
+        
+        startEngine(configBuilder);
+        awaitUntilConnectorIsReady();
+
+        // Insert 10 streaming records with non-empty name - these should pass the filter
+        for (int i = SNAPSHOT_RECORDS + STREAM_RECORDS; i < SNAPSHOT_RECORDS + STREAM_RECORDS + 10; i++) {
+            TestHelper.executeInDatabase(
+                "INSERT INTO public.test_1(id, name) VALUES (" + i + ", 'Shishir');",
+                DEFAULT_COLOCATED_DB_NAME
+            );
+        }
+
+        List<SourceRecord> records = new ArrayList<>();
+
+        if (snapshotMode.getValue().equals(SnapshotMode.INITIAL.getValue())) {
+            // INITIAL mode: Expect 10 snapshot records + 10 streaming records
+            waitAndFailIfCannotConsume(records, 20);
+            assertEquals(20, records.size(), 
+                "Expected 20 records (10 from snapshot + 10 from streaming) in INITIAL mode");
+        } else {
+            // NEVER mode: Expect only 10 streaming records
+            waitAndFailIfCannotConsume(records, STREAM_RECORDS);
+            assertEquals(STREAM_RECORDS, records.size(), 
+                "Expected 10 records (only streaming) in NEVER mode");
+        }
```

Comment:
Will this test fail, if you remove your fix and run with snapshotMode = NEVER? If not we can remove the run where we are running with NEVER mode. The INITIAL mode test run will test the streaming case as well, since it transitions to streaming after completing snapshot?